### ISSUE TITLE
Request for Comments

### ISSUE THREAD

**Issue by jscarle** [dbasoftware/freshdesksharp#2]
I'm working on implementing the full API, if you have time, I'd love any feedback or comments:
https://github.com/jscarle/Freshdesk.NET

**Comment by fyndor** [comment]
Well for one I would recommend converting your code to always use asynchronous communication.  If someone using your api wants to use one of the "hacks" to turn the async task into a synchronous call then fine but I would not provide synchronous calls in your api.  It is bad practice.  If this is being done from a client application then the gui can't continue it's message loop and will hang while the calls are being made and on a server it will just eat up CPU resources unnecessarily.

**Comment by jscarle** [comment]
I had already prepared the base framework for asynchronous support. I went through all the methods and propagated Async versions all the way up. So now the library exposes both Synchronous and Asynchronous versions of each method. For the synchronous version, they simply call the asynchronous version under the hood using WaitAndUnwrapException as the recommended best practice to do so discussed in detail here : https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c

The changes are committed. Thanks for your feedback!
